draft-ietf-webtrans-http3.md

```diff
@@ -197,6 +197,26 @@ The `webtransport` HTTP Upgrade Token uses the Capsule Protocol as defined in
 HTTP-DATAGRAM}} is not required by WebTransport and can safely be ignored by WebTransport
 endpoints.
 
+## Subprotocol Negotiation
+
+WebTransport over HTTP/3 offers a subprotocol negotiation mechanism, similar to
+TLS Application-Layer Protocol Negotiation Extension (ALPN) {{?RFC7301}}; the
+intent is to simplify porting pre-existing protocols that use QUIC and rely on
+this functionality.
+
+The user agent MAY include a `WebTransport-Subprotocols-Available` header field
+in the CONNECT request, enumerating the possible subprotocols; if received, the
+server MUST include a `WebTransport-Subprotocol` field with the subprotocol
+chosen in the 2xx response, or respond with an error code instead.
+
+Both `WebTransport-Subprotocols-Available` and `WebTransport-Subprotocol` are
+Structured Fields {{!RFC8941}}. `WebTransport-Subprotocols-Available` is a List
+of Tokens, and `WebTransport-Subprotocol` is a Token. The token in the
+`WebTransport-Subprotocol` response header field MUST be one of the tokens
+listed in `WebTransport-Subprotocols-Available` of the request.  The semantics
+of individual token values is entirely up to the WebTransport resource in
+question.
```

Comment:
```suggestion
question, and are not registered with IANA in the way that ALPN tokens are.
```

Reply:
This is technically not true. Assume I have a Foobar Protocol, and it has subprotocols `foo` and `bar`. I can create an IANA registry for subprotocols supported by the Foobar resources, and put `foo` and `bar` into it.

Reply:
```suggestion
question, and are not registered in IANA's "ALPN Protocol IDs" registry.
```